python/lsst/meas/algorithms/convertReferenceCatalog.py

```diff
@@ -64,6 +65,156 @@ def addRefCatMetadata(catalog):
+            doc="is position known?",
+        )
+    for filterName in filterNameList:
+        schema.addField(
+            field="%s_flux" % (filterName,),
+            type=numpy.float64,
+            doc="flux in filter %s" % (filterName,),
+            units="nJy",
+        )
+    for filterName in filterNameList:
+        schema.addField(
+            field="%s_fluxErr" % (filterName,),
+            type=numpy.float64,
+            doc="flux uncertainty in filter %s" % (filterName,),
+            units="nJy",
+        )
+    if addIsPhotometric:
+        schema.addField(
+            field="photometric",
+            type="Flag",
+            doc="set if the object can be used for photometric calibration",
+        )
+    if addIsResolved:
+        schema.addField(
+            field="resolved",
+            type="Flag",
+            doc="set if the object is spatially resolved",
+        )
+    if addIsVariable:
+        schema.addField(
+            field="variable",
+            type="Flag",
+            doc="set if the object has variable brightness",
+        )
+    lsst.afw.table.CovarianceMatrix2fKey.addFields(
+        schema=schema,
+        prefix="coord",
+        names=["ra", "dec"],
+        units=["rad", "rad"],
+        diagonalOnly=True,
+    )
+
+    if fullPositionInformation:
+        schema.addField(
+            field="epoch",
+            type=numpy.float64,
+            doc="date of observation (TAI, MJD)",
+            units="day",
+        )
+        schema.addField(
+            field="pm_ra",
+            type="Angle",
+            doc="proper motion in the right ascension direction = dra/dt * cos(dec)",
+            units="rad/year",
+        )
+        schema.addField(
+            field="pm_dec",
+            type="Angle",
+            doc="proper motion in the declination direction",
+            units="rad/year",
+        )
+        lsst.afw.table.CovarianceMatrix2fKey.addFields(
+            schema=schema,
+            prefix="pm",
+            names=["ra", "dec"],
+            units=["rad/year", "rad/year"],
+            diagonalOnly=True,
+        )
+        schema.addField(
+            field="pm_flag",
+            type="Flag",
+            doc="Set if proper motion or proper motion error is bad",
+        )
+        schema.addField(
+            field="parallax",
+            type="Angle",
+            doc="parallax",
+            units="rad",
+        )
+        schema.addField(
+            field="parallaxErr",
+            type="Angle",
+            doc="uncertainty in parallax",
+            units="rad",
+        )
+        schema.addField(
+            field="parallax_flag",
+            type="Flag",
+            doc="Set if parallax or parallax error is bad",
+        )
+        # Add all the off-diagonal covariance terms
+        fields = ["coord_ra", "coord_dec", "pm_ra", "pm_dec", "parallax"]
+        units = ["rad", "rad", "rad/year", "rad/year", "rad"]
+        for i in range(5):
```

Comment:
Why are you doing these loops with indices? `for field, unit in zip(itertools.combinations(fields, r=2), itertools.combinations(units, r=2)):` and then e.g. `{field[0]}_{field[1]}_Cov`, should give you all the entries, no double loop or indices necessary.

Reply:
I didn't use this because I'm not really familiar with itertools. I'm fine with changing this.